examples/common/mod.rs

```diff
@@ -73,7 +99,98 @@ impl Default for PacketStats {
         PacketStats {
             start_time: std::time::Instant::now(),
             current: 0,
-            history: heapless::HistoryBuffer::new(),
+            bytes: 0,
+            message_history: History::new(),
+            byte_history: History::new(),
+        }
+    }
+}
+
+fn log_metric(metric_name: &str, metric_value: u128, elapsed_micros: u128, history: &History) {
+    log::info!(
+        "{}: {:>8} | 1s average: {:>8}/s) | {}s average: {:>8}/s{}",
+        metric_name,
+        metric_value,
+        metric_value * 1_000_000 / elapsed_micros,
+        PACKET_STATS_WINDOW_SIZE,
+        history.as_slice().iter().sum::<u128>() * 1_000_000
+            / (history.len() as u128)
+            / elapsed_micros,
+        if history.len() < PACKET_STATS_WINDOW_SIZE {
+            " (?)"
+        } else {
+            ""
+        },
+    );
+}
+
+fn log_sys_metrics() {
+    let sys = System::new();
+
+    match sys.networks() {
+        Ok(netifs) => {
+            log::info!("Networks:");
+            for netif in netifs.values() {
+                log::info!("{} ({:?})", netif.name, netif.addrs);
+            }
+        }
+        Err(x) => log::info!("Networks: error: {}", x),
+    }
+
+    match sys.networks() {
```

Comment:
You already called it once above. Just do both inside the same `Ok(netifs)` arm.

Reply:
Yeah just copied this from example, let me clean it up